prboom2/src/p_inter.c

```diff
@@ -804,28 +802,12 @@ static void P_KillMobj(mobj_t *source, mobj_t *target)
   // Drop stuff.
   // This determines the kind of object spawned
   // during the death frame of a thing.
-
-  switch (target->type)
-    {
-    case MT_WOLFSS:
-    case MT_POSSESSED:
-      item = MT_CLIP;
-      break;
-
-    case MT_SHOTGUY:
-      item = MT_SHOTGUN;
-      break;
-
-    case MT_CHAINGUY:
-      item = MT_CHAINGUN;
-      break;
-
-    default:
-      return;
-    }
-
-  mo = P_SpawnMobj (target->x,target->y,ONFLOORZ, item);
-  mo->flags |= MF_DROPPED;    // special versions of items
+  if (target->info->droppeditem != MT_NULL)
+  {
+    mobj_t     *mo;
+    mo = P_SpawnMobj (target->x,target->y,ONFLOORZ, target->info->droppeditem);
+    mo->flags |= MF_DROPPED;    // special versions of items
+  }
 
 #ifdef GL_DOOM
   if (target->momx == 0 && target->momy == 0)
```

Comment:
I'm unsure what the MF_FOREGROUND flag is doing here for GLBOOM, do we only want it in corpses that drop stuff?

Reply:
Well, I'll just leave it with the same behavior as before.